scipy/stats/_multivariate.py

```diff
@@ -580,83 +379,96 @@ def __init__(self, seed=None):
         super().__init__(seed)
         self.__doc__ = doccer.docformat(self.__doc__, mvn_docdict_params)
 
-    def __call__(self, mean=None, cov=1, allow_singular=False, seed=None,
-                 inverse_cov=None):
+    def __call__(self, mean=None, cov=1, allow_singular=False, seed=None):
         """Create a frozen multivariate normal distribution.
 
         See `multivariate_normal_frozen` for more information.
         """
         return multivariate_normal_frozen(mean, cov,
                                           allow_singular=allow_singular,
-                                          seed=seed, inverse_cov=inverse_cov)
+                                          seed=seed)
 
-    def _process_parameters(self, mean, cov, inverse_cov, allow_singular=True):
+    def _process_parameters(self, mean, cov, allow_singular=True):
         """
         Infer dimensionality from mean or covariance matrix, ensure that
         mean and covariance are full vector resp. matrix.
         """
+        if isinstance(cov, _covariance.Covariance):
+            dim, mean, cov_object = self._process_parameters_new(mean, cov)
```

Comment:
"new" and "old" only have meaning in this PR. Future readers will more likely benefit from an intention-revealing name here. What do you think about:
```suggestion
            dim, mean, cov_object = self._process_parameters_from_cov_object(mean, cov)
```

Reply:
Sure, I can rename them. I will want to indicate in a comment that eventually I think we want to get rid of the old system, but maybe that is when we create new multivariate distribution classes to solve all of gh-16278 without the backwards compatibility nightmares.

Reply:
Renamed.